### PR TITLE
Collect process (beats) stdout and stderr

### LINKED ISSUE #88: Elastic-agent should not swallows stderr and panics
Hi Team,

this [issue](https://github.com/elastic/integrations/issues/2602#issuecomment-1023268069) proved that we need a more user-friendly method to collect all panics. Otherwise we won't be able to grab anything from customers.

In the mentioned case I used `strace` to listen what `filebeat` is sending to `/dev/null`, but we need a more trival method.

### PR DESCRIPTION
**Pull request by AndersonQ**
## What does this PR do?

The Elastic Agent now collects the `strout` and `stderr `of the (sub)process it runs.

## Why is it important?

When application crashes (panics), the "crash/panic" information is lost as they log directly to files and by default the stdout and stderr of a sub process is connected to `os.DevNull`.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- ~~[ ] I have made corresponding changes to the documentation~~
- ~~[ ] I have made corresponding change to the default configuration files~~
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## How to test this PR locally

Panic a beat and check the logs are there :)

There is the [`agent-88-collest-beats-stdout-stderr`](https://github.com/elastic/beats/tree/agent-88-collest-beats-stdout-stderr) beats branch has a modified filebeat that exposes a [HTTP server](https://github.com/elastic/beats/blob/agent-88-collest-beats-stdout-stderr/x-pack/filebeat/httppanic/httppanic.go#L18) which will [panic](https://github.com/elastic/beats/blob/46b5d90be40a31519854b4b68e09be0b002c5596/libbeat/monitoring/report/log/log.go#L168) and [log to stdout and stderr](https://github.com/elastic/beats/blob/agent-88-collest-beats-stdout-stderr/x-pack/filebeat/httppanic/httppanic.go#L23) upon request.

Just compile the agent and the beats together, run the agent with any policy that will install and run a filebeat (filebeat_monitoring also works).
Look for logs like `starting HTTP panic server on port :424X`, where X is a random number chosen at startup. The agent should already log it, but filebeat will [log it](https://github.com/elastic/beats/blob/agent-88-collest-beats-stdout-stderr/x-pack/filebeat/httppanic/httppanic.go#L36) as well.
Once you have it, you can:

```shell
curl localhost:424X # will make filebeat log to stdout and stderr
curl -X PANIC localhost:424X # will make filebeat panic
```

## Related issues

- Closes #88 

## Logs

```text
{"log.level":"info","@timestamp":"2022-05-13T16:19:28.760Z","log.origin":{"file.name":"process/stdlogger.go","file.line":50},"message":"filebeat stdout: \"starting HTTP panic server on port :4248\\n\"","agent.console.name":"filebeat","agent.console.type":"stdout","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2022-05-13T16:19:47.210Z","log.origin":{"file.name":"process/stdlogger.go","file.line":50},"message":"filebeat stdout: \"logging to stdout before PANIC panic!\\n\"","agent.console.name":"filebeat","agent.console.type":"stdout","ecs.version":"1.6.0"}
{"log.level":"error","@timestamp":"2022-05-13T16:20:01.802Z","log.origin":{"file.name":"process/stdlogger.go","file.line":50},"message":"filebeat stderr: \"panic: HTTP Panic server said to panic!\\n\\ngoroutine 65 [running]:\\ngithub.com/elastic/beats/v7/libbeat/monitoring/report/log.(*reporter).snapshotLoop(0xc000724640)\\n\\t/go/src/github.com/elastic/beats/libbeat/monitoring/report/log/log.go:169 +0x67d\\ngithub.com/elastic/beats/v7/libbeat/monitoring/report/log.MakeReporter.func1()\\n\\t/go/src/github.com/elastic/beats/libbeat/monitoring/report/log/log.go:134 +0x5a\\ncreated by github.com/elastic/beats/v7/libbeat/monitoring/report/log.MakeReporter\\n\\t/go/src/github.com/elastic/beats/libbeat/monitoring/report/log/log.go:132 +0x23e\\n\"","agent.console.name":"filebeat","agent.console.type":"stderr","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2022-05-20T10:31:02.518Z","log.origin":{"file.name":"process/stdlogger.go","file.line":50},"message":"filebeat stdout: \"starting HTTP panic server on port :4247\\n\"","agent.console.name":"filebeat","agent.console.type":"stdout","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2022-05-20T10:31:02.663Z","log.origin":{"file.name":"process/stdlogger.go","file.line":50},"message":"filebeat_monitoring stdout: \"starting HTTP panic server on port :4249\\n\"","agent.console.name":"filebeat_monitoring","agent.console.type":"stdout","ecs.version":"1.6.0"}
```

Screenshot from 2022-05-20 12-22-41

### CODE REVIEW

File: internal/pkg/core/plugin/process/stdlogger_test.go [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+				appName string
+				logTo   logStd
+			}{
+				appName: "somebeats",
+				logTo:   logStdOut,
+			},
+			logMsg:   "stdout log",
+			logLevel: zapcore.InfoLevel,
+		},
+		{name: "capture stderr",
```
Comment (aleksmaus) [inline internal/pkg/core/plugin/process/stdlogger_test.go:34]: nit: maybe alight the ```name``` with other properties in the structure initialization, here and above

File: internal/pkg/core/plugin/process/status.go [hunk 2]
```diff
@@ -101,9 +101,7 @@ func (a *Application) restart(proc *process.Info) {
 	a.stopWatcher(proc)
 
 	// kill the process
-	if proc != nil && proc.Process != nil {
-		_ = proc.Process.Kill()
-	}
+	_ = proc.Process.Kill()
```
Comment (aleksmaus) [inline internal/pkg/core/plugin/process/status.go:104]: so, there is no way that the proc or the proc.Process can be nil here? 
seeing that the nil check was here before and now is removed.

**Review by aleksmaus — APPROVED** [review]
LGTM

File: internal/pkg/core/plugin/process/status.go (hunk 2 above)
Comment (ph) [reply internal/pkg/core/plugin/process/status.go:104]: That seems indeed odd to me, I would keep that and separate investigation. See https://github.com/elastic/beats/pull/26126

**Review by ph — CHANGES_REQUESTED** [review]
@AndersonQ Only a few things:

- Add a changelog entry.
- Revert the proc.kill
- I would fix the alignment.

Other than the above it looks good to me.

**Review by ph — APPROVED** [review]

**Pull request merged**